homeassistant/components/home_plus_control/__init__.py

```diff
@@ -0,0 +1,177 @@
+    """Set up Legrand Home+ Control from a config entry."""
+    hass_entry_data = hass.data[DOMAIN].setdefault(config_entry.entry_id, {})
+
+    # Retrieve the registered implementation
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, config_entry
+        )
+    )
+
+    # Using an aiohttp-based API lib, so rely on async framework
+    # Add the API object to the domain's data in HA
+    hass_entry_data[API] = api.HomePlusControlAsyncApi(
+        hass, config_entry, implementation
+    )
+
+    # Set of entity unique identifiers of this integration
+    hass_entry_data[ENTITY_UIDS] = set()
+
+    # Integration dispatchers
+    hass_entry_data[DISPATCHER_REMOVERS] = []
+
+    # Register the Data Coordinator with the integration
+    async def async_update_data():
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                module_data = await hass_entry_data[API].fetch_data()
+        except HomePlusControlApiError as err:
+            raise UpdateFailed(
+                f"Error communicating with API: {err} [{type(err)}]"
+            ) from err
+
+        # Remove obsolete entities from Home Assistant
+        entity_uids_to_remove = hass_entry_data[ENTITY_UIDS] - set(module_data)
+        device_registry = hass.helpers.device_registry.async_get(hass)
+        for uid in entity_uids_to_remove:
+            hass_entry_data[ENTITY_UIDS].remove(uid)
+            device = device_registry.async_get_device({(DOMAIN, uid)})
+            device_registry.async_remove_device(device.id)
+
+        # Send out signal for new entity addition to Home Assistant
+        new_entity_uids = set(module_data) - hass_entry_data[ENTITY_UIDS]
+        hass_entry_data[ENTITY_UIDS].update(new_entity_uids)
+        if len(new_entity_uids) > 0:
+            dispatcher.async_dispatcher_send(
+                hass,
+                SIGNAL_ADD_ENTITIES,
+                new_entity_uids,
+                coordinator,
+            )
+
+        return module_data
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="home_plus_control_module",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=60),
+    )
+    hass_entry_data[DATA_COORDINATOR] = coordinator
+
+    # Continue setting up the platform
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
```

Comment:
We make sure the platforms are loaded before refreshing the coordinator.
```suggestion
    async def start_platforms():
        """Continue setting up the platforms."""
        await asyncio.gather(
            *[
                hass.config_entries.async_forward_entry_setup(config_entry, platform)
                for platform in PLATFORMS
            ]
        )
        # Only refresh the coordinator after all platforms are loaded.
        await coordinator.async_refresh()
        
    hass.async_create_task(start_platforms())
```